internal/scms/bitbucket_cloud/commit_status.go

```diff
@@ -0,0 +1,102 @@
+package bitbucket_cloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ktrysmt/go-bitbucket"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/argoproj-labs/gitops-promoter/api/v1alpha1"
+	"github.com/argoproj-labs/gitops-promoter/internal/metrics"
+	"github.com/argoproj-labs/gitops-promoter/internal/scms"
+	"github.com/argoproj-labs/gitops-promoter/internal/utils"
+)
+
+// CommitStatus implements the scms.CommitStatusProvider interface for Bitbucket Cloud.
+type CommitStatus struct {
+	client    *bitbucket.Client
+	k8sClient client.Client
+}
+
+var _ scms.CommitStatusProvider = &CommitStatus{}
+
+// NewBitbucketCloudCommitStatusProvider creates a new instance of CommitStatus for Bitbucket Cloud.
+func NewBitbucketCloudCommitStatusProvider(k8sClient client.Client, secret v1.Secret, domain string) (*CommitStatus, error) {
+	client, err := GetClient(secret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CommitStatus{client: client, k8sClient: k8sClient}, nil
+}
+
+// Set sets the commit status for a given commit SHA in the specified repository.
+func (cs *CommitStatus) Set(ctx context.Context, commitStatus *v1alpha1.CommitStatus) (*v1alpha1.CommitStatus, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Setting Commit Phase")
+
+	repo, err := utils.GetGitRepositoryFromObjectKey(ctx, cs.k8sClient, client.ObjectKey{
+		Namespace: commitStatus.Namespace,
+		Name:      commitStatus.Spec.RepositoryReference.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get repo: %w", err)
+	}
+
+	commitOptions := &bitbucket.CommitsOptions{
+		Owner:    repo.Spec.BitbucketCloud.Workspace,
+		RepoSlug: repo.Spec.BitbucketCloud.Repository,
+		Revision: commitStatus.Spec.Sha,
+	}
+
+	commitStatusOptions := &bitbucket.CommitStatusOptions{
+		State:       phaseToBuildState(commitStatus.Spec.Phase),
+		Key:         commitStatus.Spec.Name,
+		Url:         commitStatus.Spec.Url,
```

Comment:
@ushiradineth 
From what I saw testing it looks like the URL is empty and bitbucket does not allow that it is a required parameter
@zachaller does it need to be the url or having `example.org` will work just fine?

Reply:
@ushiradineth Actually an empty URL won't work we will also need https://github.com/ushiradineth/gitops-promoter/pull/5/files